src/overlays/actors/ovl_Bg_Spout_Fire/z_bg_spout_fire.c

```diff
@@ -31,39 +34,172 @@ const ActorInit Bg_Spout_Fire_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A60CDC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A60D10.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A60DA0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A60E08.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/BgSpoutFire_Update.s")
+static TexturePtr D_80A61194[] = {
+    object_fwall_Tex_003CA0, object_fwall_Tex_0040A0, object_fwall_Tex_0044A0, object_fwall_Tex_0048A0,
+    object_fwall_Tex_004CA0, object_fwall_Tex_0050A0, object_fwall_Tex_0054A0, object_fwall_Tex_0058A0,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Spout_Fire/func_80A61040.s")
+static s32 texturesDesegmented = 0;
+
+void BgSpoutFire_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 i;
+    BgSpoutFire* this = THIS;
+
+    this->actor.scale.z = 0.13499999f;
+    this->actor.scale.x = 0.13499999f;
+    this->actor.scale.y = 0.01f;
+    this->unk_14A = 0;
+    Collider_InitAndSetCylinder(globalCtx, &this->unk14C, &this->actor, &sCylinderInit);
+    this->unk14C.dim.pos.y = this->actor.world.pos.y;
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    if (!texturesDesegmented) {
+        for (i = 0; i < ARRAY_COUNT(D_80A61194); i++) {
+            D_80A61194[i] = Lib_SegmentedToVirtual(D_80A61194[i]);
+        }
+        texturesDesegmented = true;
+    }
+    this->actionFunc = func_80A60C94;
+}
+
+void BgSpoutFire_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpoutFire* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->unk14C);
+}
+
+s32 func_80A60C24(BgSpoutFire* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f sp18;
+
+    Actor_OffsetOfPointInActorCoords(&this->actor, &sp18, &player->actor.world.pos);
+    if ((fabsf(sp18.x) < 100.0f) && (fabsf(sp18.z) < 120.0f)) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void func_80A60C94(BgSpoutFire* this, GlobalContext* globalCtx) {
+    if (func_80A60C24(this, globalCtx)) {
+        this->actor.draw = BgSpoutFire_Draw;
+        this->unk_148 = 5;
+        this->actionFunc = func_80A60CDC;
+    }
+}
+
+void func_80A60CDC(BgSpoutFire* this, GlobalContext* globalCtx) {
+    if (this->unk_148 != 0) {
+        this->unk_148--;
+    }
+    if (this->unk_148 == 0) {
+        this->actionFunc = func_80A60D10;
+    }
+}
+
+void func_80A60D10(BgSpoutFire* this, GlobalContext* globalCtx) {
+    if (func_80A60C24(this, globalCtx)) {
+        Math_StepToF(&this->actor.scale.y, 0.1f, 0.024999999f);
+        return;
+    }
+    if (Math_StepToF(&this->actor.scale.y, 0.01f, 0.01f)) {
+        this->actor.draw = NULL;
+        this->actionFunc = func_80A60C94;
+        return;
+    }
+    this->unk_148 = 0;
+}
+
+void func_80A60DA0(Actor* thisx, GlobalContext* globalCtx) {
```

Comment:
Out of curiosity, any reason you chose to use the base actor here instead of `BgSpoutFire* this`? I don't see any particular reason why that would be the case.

Reply:
otherwise i'm getting warnings by calling the function in other places

Reply:
Is it called/used any where else besides `BgSpoutFire_Update`? If not you can just change that to pass in `this` instead of `&this->actor`.